### PR TITLE
Add search to homepage

### PR DESCRIPTION
**Pull request by pawalt**
This PR:
- Adds search to the homepage
- Adds tabs for downtowns and users on the homepage

### CODE REVIEW

**Review by ArmaanT — DISMISSED** [review]
I would add a button (or some other way to clear the current search), as well as a number next to the tabs to tell how many results for a search there are.

File: main/views.py [hunk 1]
```diff
@@ -26,16 +27,33 @@ class HomeView(TemplateView):
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        downtowns = Downtown.objects.all()
+        query = self.request.GET.get('query', '')
+        downtowns = Downtown.objects.filter(name__contains=query)
+        users = User.objects.filter(
```
Comment (ArmaanT) [inline main/views.py]: Use Django's get_user_model instead of directly referring to the user model. That the code works even if the user class changes.

**Comment by pawalt** [comment]
I added a result count, but I didn't add a text clearer. While it would make sense on mobile, I don't think it's any easier to click a clear button than to just press the backspace button on desktop.

**Comment by ArmaanT** [comment]
LGTM!

**Pull request merged**